### ISSUE TITLE
MediaUriElement.MediaPosition does not get reset to 0 when changing source.

### ISSUE THREAD

**Issue by perkalkyl** [Sascha-L/WPF-MediaKit#39]
Hi!
Just started to implement your library to get rid of the memoryleak in the original MediaElement. Fantastic work! Thank you so much for your efforts. I would gladly participate in your project in the small way I can. 
My first little contribution is this probable bug report.
After playing a source, and changing to another video, the MediaPosition property does not get reset to 0.
My handling of closing the previous (and still playing) source is:

                try
                {
                    if (CurrentPlayer.Source != null)
                    {
                        CurrentPlayer.Stop();
                        CurrentPlayer.Source = null;
                        CurrentPlayer.MediaPosition = 0; //Added to reset the MediaPosition value.
                    }
                }
                catch (Exception ex2)
                {
                    WriteLog("ERROR: " + ex2.Message + "   while in PlayNext(){ CurrentPlayer.Stop; CurrentPlayer.Source=null;}");

                }

If I omit the "CurrentPlayer.MediaPosition = 0;" line, the MediaElement will have the last shown position as current position when the  CurrentPlayer_MediaOpened(object sender, RoutedEventArgs e) event is triggered as a result of the next source being opened. It doesn't help to try to set the MediaPosition to 0 in the event. It has to be done before changing source.

note: CurrentPlayer is a Property to get the currently active MediaUriElement in the application. I have tested this on a "clean" test app with the same result.

With warm regards
Per

**Comment by skaterlui** [comment]
Hi xmedeko,

maybe the MediaElementBase.Stop() method can be overridden in the MediaSeekingPlayer.cs to set the MediaPosition property there to 0.

**Comment by xmedeko** [comment]
That would change the behaviour. Currently, the Stop does not change the `MediaPosition`, so Play after stop starts from the last position. If you set `MediaPosition` to 0, then the Stop will move to the beginning of the video. It may be desirable behaviour for some users, but maybe not for all. Open another issue if you want to change the Stop behaviour.

IMO the bug is in the `MediaSeekingElement.OnMediaPlayerOpened`, I'll fix it.

**Comment by skaterlui** [comment]
Hi xmedeko,
thanks for the reply. I don't need this feature at all. I'm fine the way it behaves. It was just a suggestion for perkalkyls issue. Meanwhile I tried a view players. 
-VLC
-Media Player Classic
-Windows Media Player
-LEAWO BluRay Player
They are all setting the mediaposition to 0 when you hit the stop button. The pause button does it the way you mentioned, resuming the video at the last mediaposition.
